cdc/puller/ddl_puller.go

```diff
@@ -442,20 +415,46 @@ func (p *ddlJobPullerImpl) handleJob(job *timodel.Job) (skip bool, err error) {
 		return true, errors.Trace(err)
 	}
 
+	defer func() {
```

Comment:
Why did you move this defer block here? It can't cover the situation that would happen at line 413.